v1/compat.go

```diff
@@ -0,0 +1,87 @@
+package machinery
+
+import (
+	"context"
+
+	opentracing "github.com/opentracing/opentracing-go"
+
+	"github.com/RichardKnop/machinery/v1/tasks"
+	"github.com/RichardKnop/machinery/v1/tracing"
+)
+
+var (
+	// These ensure opentracing is automatically called to preserve backwards-compatible behavior
+	// TODO: deprecate, or expose
+	defaultPreTaskHandler         = OpenTracingPreTaskHandler
+	defaultPostTaskHandler        = OpenTracingPostTaskHandler
+	defaultPrePublishHandler      = OpenTracingPrePublishTaskHandler
+	defaultPreGroupPublishHandler = OpenTracingPreGroupPublishTaskHandler
+	defaultPreChordPublishHandler = OpenTracingPreChordPublishTaskHandler
+	defaultPreChainPublishHandler = OpenTracingPreChainPublishTaskHandler
+)
+
+// finishFun is used by the prepublish handlers on the server
+// to ensure a function that returns nil doesn't panic
+// is this overkill for panics?
+func finishFunc(funcs ...func()) func() {
+	return func() {
+		for _, f := range funcs {
+			f := f // pin
+			if f != nil {
+				f()
+			}
+		}
+	}
+}
+
+// OpenTracingPreTaskHandler starts an opentracing span based on incoming headers
+func OpenTracingPreTaskHandler(ctx context.Context, signature *tasks.Signature) context.Context {
+	// try to extract trace span from headers and add it to the function context
+	// so it can be used inside the function if it has context.Context as the first
+	// argument. Start a new span if it isn't found.
+	taskSpan := tracing.StartSpanFromHeaders(signature.Headers, signature.Name)
+	tracing.AnnotateSpanWithSignatureInfo(taskSpan, signature)
+	return opentracing.ContextWithSpan(ctx, taskSpan)
+}
+
+// OpenTracingPostTaskHandler ends an opentracing span based on the passed in context
+func OpenTracingPostTaskHandler(ctx context.Context, _ *tasks.Signature) {
+	taskSpan := opentracing.SpanFromContext(ctx)
+	if taskSpan != nil {
+		taskSpan.Finish()
+	}
+}
+
+// OpenTracingPrePublishTaskHandler propagates opentracing information in outgoing task header information
+func OpenTracingPrePublishTaskHandler(ctx context.Context, signature *tasks.Signature) func() {
+	span, _ := opentracing.StartSpanFromContext(ctx, "SendTask", tracing.ProducerOption(), tracing.MachineryTag)
+
+	// tag the span with some info about the signature
+	signature.Headers = tracing.HeadersWithSpan(signature.Headers, span)
```

Comment:
No need for `signature != nil` check here?

Reply:
It would likely have panic-ed earlier, I think.